<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>537</width>
    <height>334</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="tabPosition">
      <enum>QTabWidget::North</enum>
     </property>
     <property name="tabShape">
      <enum>QTabWidget::Triangular</enum>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>CTRL</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Enable</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="comboBox">
         <property name="currentIndex">
          <number>1</number>
         </property>
         <item>
          <property name="text">
           <string>0: Disable motor</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1: Enable motor</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Current Gain</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="comboBox_2">
         <property name="currentIndex">
          <number>3</number>
         </property>
         <item>
          <property name="text">
           <string>00: Gain of 5V/V</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>01: Gain of 10V/V</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>10: Gain of 20V/V</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>11: Gain of 40V/V</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Dead Set Time</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="comboBox_3">
         <item>
          <property name="text">
           <string>00: 410-ns dead time</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>01: 460-ns dead time</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>10: 670-ns dead time</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>11: 880-ns dead time</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>TORQUE</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="1">
        <widget class="QSpinBox" name="spinBox">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="value">
          <number>255</number>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Torque</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>I = </string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>I=(2.75*TORQUE)/(256*ISGAIN*7.5E-3)</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDial" name="dial">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="value">
          <number>255</number>
         </property>
         <property name="notchTarget">
          <double>5.000000000000000</double>
         </property>
         <property name="notchesVisible">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QCheckBox" name="checkBox">
         <property name="text">
          <string>Auto Send</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>OFF</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Fixed time off</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="spinBox_2">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="value">
          <number>48</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Increments of 525ns. From 525ns to 133.8uS</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>t_off = 25.725uS</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>BLANK</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_3">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Current Trip Blanking Time</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="spinBox_3">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="value">
          <number>128</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Increments of 20ns. From 500nS to 5.14uS</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>Also sets the minimum on-time of PWM</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>Tblank = 3.06nS</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_5">
      <attribute name="title">
       <string>DECAY</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_4">
       <item row="0" column="0">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Mixed Decay trans time</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="spinBox_4">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="value">
          <number>16</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="label_16">
         <property name="text">
          <string>Increments of 525nS</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Decay Mode</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QComboBox" name="comboBox_4">
         <item>
          <property name="text">
           <string>000: Force slow decay at all times</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>010: Force fast decay at all times</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>011: Used Mixed decay at all times</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>101: Use auto mixed decay at all times</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>T_Decay = </string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_6">
      <attribute name="title">
       <string>DRIVE</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_5">
       <item row="0" column="0">
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>Over Current Protection Threshold</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="comboBox_5">
         <property name="currentIndex">
          <number>1</number>
         </property>
         <item>
          <property name="text">
           <string>00: 250mV</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>01: 500mV</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>10: 750mV</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>11: 1000mV</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>Over Current Protection Deglitch Time</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="comboBox_6">
         <property name="currentIndex">
          <number>1</number>
         </property>
         <item>
          <property name="text">
           <string>00: 1.05 uS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>01: 2.1 5uS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>10: 4.2 uS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>11: 8.4 uS</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_20">
         <property name="text">
          <string>Gate drive sink time</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="comboBox_7">
         <property name="currentIndex">
          <number>2</number>
         </property>
         <item>
          <property name="text">
           <string>00: 236 nS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>01: 525 nS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>10: 1.05 uS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>11: 2.10 uS</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>Gate drive source time</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_22">
         <property name="text">
          <string>Gate drive sink current</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QComboBox" name="comboBox_9">
         <property name="currentIndex">
          <number>3</number>
         </property>
         <item>
          <property name="text">
           <string>00: 100-mA peak (sink)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>01: 200-mA peak (sink)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>10: 300-mA peak (sink)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>11: 400-mA peak (sink)</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_23">
         <property name="text">
          <string>Gate drive source current</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QComboBox" name="comboBox_10">
         <property name="currentIndex">
          <number>3</number>
         </property>
         <item>
          <property name="text">
           <string>00: 50-mA peak (source)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>01: 100-mA peak (source)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>10: 150-mA peak (source)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>11: 200-mA peak (source)</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QComboBox" name="comboBox_19">
         <property name="currentIndex">
          <number>2</number>
         </property>
         <item>
          <property name="text">
           <string>00: 236 nS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>01: 525 nS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>10: 1.05 uS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>11: 2.10 uS</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_7">
      <attribute name="title">
       <string>STATUS</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="3" column="0">
        <widget class="QLabel" name="label_26">
         <property name="text">
          <string>Motor 2 over current</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="label_27">
         <property name="text">
          <string>Normal Operation</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_28">
         <property name="text">
          <string>Normal Operation</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="label_29">
         <property name="text">
          <string>Normal Operation</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_30">
         <property name="text">
          <string>Motor 1 predriver fault</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_31">
         <property name="text">
          <string>Motor 2 predriver fault</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>Update Values</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_24">
         <property name="text">
          <string>Over temperature Shutdown</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_25">
         <property name="text">
          <string>Motor 1 over current</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_32">
         <property name="text">
          <string>Undervoltage lockout</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLabel" name="label_33">
         <property name="text">
          <string>Normal Operation</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QLabel" name="label_34">
         <property name="text">
          <string>Normal Operation</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QLabel" name="label_35">
         <property name="text">
          <string>Normal Operation</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Controls</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="send_button">
        <property name="text">
         <string>Send Values to UDrive</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="reset_button">
        <property name="text">
         <string>Reset defaults</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>dial</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>392</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>392</x>
     <y>43</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>dial</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>392</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>392</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
